Python/Include/Range.hpp

```diff
@@ -56,6 +58,9 @@ namespace Rover {
 
   template<typename Begin, typename End, typename Granularity>
   void export_range(pybind11::module& module, std::string_view type_name) {
+    static_assert(!std::is_same_v<Begin, pybind11::object> &&
```

Comment:
This one is necessary here as of now as the template version of export_range relies on C++ boxing. Mixing C++ types and py objects is more complicated as I need to ensure both kinds of boxing are done properly. Let me know if I need to pursue that.